admin/ajax.php

```diff
@@ -151,8 +151,7 @@ function wpseo_ajax_replace_vars() {
 	$wp_query->queried_object_id = $post->ID;
 
 	$omit = array( 'excerpt', 'excerpt_only', 'title' );
-	echo wpseo_replace_vars( stripslashes( filter_input( INPUT_POST, 'string' ) ), $post, $omit );
-	die;
+	wpseo_ajax_json_echo_die( wp_strip_all_tags( wpseo_replace_vars( stripslashes( filter_input( INPUT_POST, 'string' ) ), $post, $omit ) ) );
```

Comment:
This output differs from the current output.
It will convert to a string in the `json_encode` function, resulting in double quotes being placed around the output.

Current output is just plain data.

Could you remove the `wpseo_ajax_json_echo_die` and just do an `echo` and `die` like it was before?

Reply:
@moorscode Then the output would still not be properly escaped. Output in response to an ajax call should be encode with `wp_json_encode()`.

I'll have another look at this (low prio).